slang/core_test.go

```diff
@@ -0,0 +1,91 @@
+package slang_test
```

Comment:
Shouldn't tests live in the `slang` package?  My understanding is that the compiler doesn't include code from `*_test.go` files when `go build` or `go run` are invoked, only `go test`.

Reply:
I prefer to use `_test` package as much as possible for following reasons:
* Ideally all private code should reachable through the public interface of the package. Keeping tests in separate `_test` package allows you to ensure this.
* When using this pattern, you have access to only public interface of the package. So the tests are very similar to how a user of the package would use it. This allows to always keep in mind the public API of the package. Also, tests can now be used as reference also by users without really having to dig into internals.

Reply:
Fine by me - those are sensible reasons (especially the second point) 😃